### PR TITLE
Add electron-chromedriver

### PR DESCRIPTION
**Pull request by liam-murphy14**
## Description of changes

add a wrapper for the electron-built chromedriver binaries for all nixpkgs supported versions of electron (30, 29, 28, 27, 24). This allows NixOS users to easily install chromedriver binaries that are compatible with specific electron versions, for use during [electron automated testing](https://www.electronjs.org/docs/latest/tutorial/automated-testing).

See [the electron github page](https://github.com/electron/electron) as well as the [electron-chromedriver NPM package](https://www.npmjs.com/package/electron-chromedriver) for more details.

Notes
1. this chromedriver binary is not already included in the release tarballs of electron/electron-bin packages
2. I modeled the nix code here after the electron-bin package, so shout out to the maintainers for providing a super nice template :)

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by yayayayaka — COMMENTED** [review]
Thanks for your contribution to nixpkgs! :purple_heart: 

Does the version scheme follow the same [release timeline as electron itself](https://www.electronjs.org/docs/latest/tutorial/electron-timelines#timeline)? If so, I would suggest to drop electron-chromedriver_{24,27} for EOL reasons and perhaps v28 as well since it will become EOL in less than a month as well.

Will do a more thorough review later.

**Comment by liam-murphy14** [comment]
Hi @yayayayaka, thanks for the review !!

Yes, this will follow the same release schedule as electron. I included 24,27,28 in order to match the provided versions of the electron and electron-bin nixpkgs derivations. I am happy to remove them, though, and just keep 29,30.

Let me know what you would prefer, I see that you are the electron package maintainer, so whatever you think is better is good with me.

Thanks again!

**Comment by liam-murphy14** [comment]
Hey @yayayayaka, I removed versions 24,27,28 at your recommendation. Let me know if you get a chance to look at this or if you need any more info from me. Thanks !!

File: pkgs/development/tools/electron/chromedriver/generic.nix
Comment (yayayayaka) [inline pkgs/development/tools/electron/chromedriver/generic.nix:18]:
```suggestion
    description = "WebDriver server for running Selenium tests on Chrome";
```

**Review by yayayayaka — COMMENTED** [review]
Thank you for keeping this PR updated and sorry that I did not find the energy to review it sooner. :(

If you don't mind, I would like to push another commit to extend the functionality of electron's `update.py` script to support updating `electron-chromedriver` as well.

**Comment by liam-murphy14** [comment]
Hey @yayayayaka, thank you for the review !! Appreciate all the suggestions, I will add address them all and push the commit in a few mins. 

Also with regard to

> If you don't mind, I would like to push another commit to extend the functionality of electron's update.py script to support updating electron-chromedriver as well.

This sounds like a great idea ! Let me know if this is something you are planning on implementing + committing, otherwise I can look into updating the `update.py` script sometime this weekend.

Cheers !
-Liam

**Comment by yayayayaka** [comment]
Result of `nixpkgs-review pr 309387` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>electron-chromedriver</li>
    <li>electron-chromedriver_29</li>
    <li>electron-chromedriver_31</li>
  </ul>
</details>

**Review by yayayayaka — APPROVED** [review]
Implemented updating chromedriver with `update.py`. @liam-murphy14, if you are fine with the diff, I would hit merge soon.

**Comment by liam-murphy14** [comment]
Hi @yayayayaka, thanks for adding that in `update.py`, that is super helpful !! I am all good with the diff. I don't have write access, so I can't merge the PR, can you ??

Thanks again !

**Comment by teutat3s** [comment]
Any objections to backporting this to 24.05? This would allow easier future backports via labels.

**Comment by liam-murphy14** [comment]
Hey @teutat3s, sure would be happy to backport this to 24.0.5 ! Just approved [your PR](https://github.com/NixOS/nixpkgs/pull/329892). Let me know if there is anything else I need to do to backport, I will have free time this weekend to work on it if needed.

Thanks !

**Pull request merged**
